File: gpt-3.5-finetuned/stage2/serial_construct_reduction_clause.f90
!T1:serial construct reduction clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        INTEGER :: errors = 0
        REAL(8),DIMENSION(LOOPCOUNT):: a, b, c
        REAL(8):: reduction_host_result
        INTEGER :: x

        SEEDDIM(1) = 1
        a = 0
        b = 0
        c = 0

        !$acc data copyin(a(1:LOOPCOUNT), b(1:LOOPCOUNT)) copyout(c(1:LOOPCOUNT))
          !$acc serial loop reduction(+:reduction_host_result)
          DO x = 1, LOOPCOUNT
            c(x) = a(x) + b(x)
            reduction_host_result = reduction_host_result + c(x)
          END DO
        !$acc end data

        DO x = 1, LOOPCOUNT
          IF (abs(c(x) - (a(x) + b(x))) .gt. PRECISION) THEN
            errors = errors + 1
          END IF
          IF (abs(reduction_host_result - DATAVAL) .gt. PRECISION) THEN
            errors = errors + 1
          END IF
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM

File: gpt-3.5-finetuned/stage2/serial_construct_reduction_clause.cpp
#include "acc_testsuite.h"
#ifndef T1
//T1:serial,construct-reduction,V:2.7-3.3
int test1(){
    int err = 0;
    srand(SEED);
    real_t * a = new real_t[1024*1024];
    real_t reduction;

    for (int x = 0; x < 1024*1024; ++x){
        a[x] = rand() / (real_t)(RAND_MAX / 10);
    }

    #pragma acc serial copyin(a[0:1024*1024]) reduction(+:reduction)
    {
        #pragma acc loop
        for (int x = 0; x < 1024*1024; ++x){
            reduction = reduction + a[x];
        }
    }

    real_t host_reduction = 0.0;
    for (int x = 0; x < 1024*1024; ++x){
        host_reduction = host_reduction + a[x];
    }
    if (fabs(host_reduction - reduction) > PRECISION){
        err += 1;
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}